### PR TITLE
test(jobs): drive the sync-job tests through a real KeaClient

### PR DESCRIPTION
**Pull request by marcinpsk**
De-mocks the `KeaIpamSyncJob` tests. They patched three whole `KeaClient` methods (`command`, `lease_get_all`, `reservation_get_page`), so the real pagination and response parsing never ran — a broken parser could stay green behind the method-level fakes.

Continues the same HTTP-boundary approach as PR #97: use a real `KeaClient` and stub only `requests.Session.post` via `stub_kea`.

## Changes (`netbox_kea/tests/test_jobs.py` only)
- **`_patch_kea`** rewritten to register the commands the job actually issues — `config-get`, `lease{4,6}-get-page`, `reservation-get-page` — so the real `lease_get_all` / `reservation_get_page` cursor loops and `command()` parsing execute.
- The seven inline data-injection overrides became `responses=` entries:
  - **Truncation** is driven by the real `max_leases` cap — two leases with a cap of 1 genuinely exceed it, so `lease_get_all` drops the overflow and returns `truncated=True` (real truncation, not a faked flag or a complete-dataset edge case; the test also asserts the overflow lease is dropped).
  - **Reservation pagination** is driven by the real `(from, source-index)` cursor via a callable stub branching on `arguments.from`.
  - **Kea error / host-cmds-absent / transport-error** paths register a result code or an exception instance at the HTTP boundary, exercising the real `command()` result-code check and `KeaException` wrapping.

## Kept as legitimate boundaries (not de-mocked)
- `test_views_leases.py:3596` narrow `command` patch — its `resp[0]`-not-a-dict guard is unreachable through the real checked `command` (`check_response` `TypeError`s first).
- The NetBox-side `sync_*` / `cleanup_stale_ips_batch` mocks and the `_make_job` / `_make_server` stand-ins.

## Verification
- Red-green proof: temporarily renaming the `lease{v}-get-page` command makes the lease tests **fail** (the old whole-method patch masked exactly that regression). Reverted.
- Full `netbox_kea/tests/` suite: **2294 passed, 2 subtests passed**.
- ruff + pre-commit hooks (incl. mock-discipline) pass.

## Summary by CodeRabbit

* **Tests**
  * Improved integration test coverage for Kea synchronization scenarios.
  * Added realistic validation for paginated lease results, response errors, transport failures, and custom configuration responses.
  * Updated stale-IP cleanup tests to verify that cleanup is skipped when lease retrieval is truncated.

### CODE REVIEW

File: netbox_kea/tests/test_jobs.py [hunk 1]
```diff
@@ -637,20 +651,15 @@ def test_subnet_prefix_map_masks_lease_from_config_get(self):
             sync_ip_ranges_enabled=False,
         )
         lease = {"ip-address": "10.0.0.50", "hostname": "masked-host", "subnet-id": 1}
+        config_with_subnet = {
```
Comment (marcinpsk) [inline netbox_kea/tests/test_jobs.py:654]: @coderabbitai

> netbox_kea/tests/test_jobs.py:636-667 — Reuse the shared `_subnet_get` helper instead of an inline `config-get` payload.

Declined — `_subnet_get` models a different Kea command. It builds a `subnet{v}-get` response: `{"arguments": {"subnet4": [{"id": …, "pools": …}]}}` (kea_stub.py:165-174). This test registers the **`config-get`** command, whose response nests subnets under `arguments.Dhcp4.subnet4` and carries the `"subnet"` CIDR field — which is exactly what the test asserts drives the `/24` mask: `_fetch_kea_subnets` reads `arguments.Dhcp{v}.subnet{v}` (jobs.py:352-363) and `_build_subnet_prefix_map` computes the prefix length from each subnet's `"subnet"` CIDR (jobs.py:386). `_subnet_get` has no `Dhcp4` wrapper and no `"subnet"` field, so substituting it would drop the subnet from the prefix map and the lease would regress to the `/32` fallback, failing the test. There is no shared `config-get` builder in kea_stub.py (only `_res_page`/`_res_get`/`_subnet_get`/`_subnet_list`), so the inline literal is the clearest option here.
